src/main/scala/nl.knaw.dans.easy.bagstore/component/FileSystemComponent.scala

```diff
@@ -120,8 +115,7 @@ trait FileSystemComponent extends DebugEnhancedLogging {
           Failure(IncompleteItemUriException("base-uri by itself is not an item-uri"))
         else {
           val uuidStr = formatUuidStrCanonically(itemIdPath.getName(0).toString.filterNot(_ == '-'))
-          assertUuidValid(uuidStr)
-          val bagId = BagId(UUID.fromString(uuidStr))
+          val bagId = BagId(getUUID(uuidStr))
```

Comment:
You don't catch the error thrown by `getUUID` in this function, right? Please test if this is a problem (perhaps this is already tested, I don't know...)

Reply:
In the original version `assert` method could also throw an exception.  But it was indeed so that the exception was not caught.  I have changed it now and also added a unit test for this case.